### PR TITLE
Migrate Manifest, FormatVersion and LocationProvider files in Core to JUnit5

### PR DESCRIPTION
**Pull request by tomtongue**
Migrate the following "Read" in iceberg-core to JUnit 5 and AssertJ style for https://github.com/apache/iceberg/issues/9085.

## Current Progress 
- [x] `TestManifestCaching.java`
- [x] `TestManifestCleanup.java`
- [x] `TestManifestEncryption.java`
- [x] `TestManifestListVersions.java`
- [x] `TestManifestReader.java` (already uses JUnit5)
- [x] `TestManifestReaderStats.java`
- [x] `TestManifestWriter.java`
- [x] `TestManifestWriterVersions.java`

Additional classes:
- [x] `TestFormatVersions.java`
- [x] `TestLocationProvider.java`

### CODE REVIEW

**Comment by tomtongue** [comment]
@nastra Migrate additional classes to JUnit5. Could you review when you have time?

File: core/src/test/java/org/apache/iceberg/TestManifestWriter.java [hunk 1]
```diff
@@ -89,48 +80,44 @@ public void testManifestPartitionStats() throws IOException {
...
-        Integer.valueOf(1),
-        Conversions.fromByteBuffer(Types.IntegerType.get(), partitionFieldSummary.lowerBound()));
-    Assert.assertEquals(
-        "Upper bound should match",
-        Integer.valueOf(3),
-        Conversions.fromByteBuffer(Types.IntegerType.get(), partitionFieldSummary.upperBound()));
+    assertThat(partitionFieldSummary.containsNull()).isFalse();
+    assertThat(partitionFieldSummary.containsNaN()).isFalse();
+    assertThat(
+            (Integer)
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestManifestWriter.java:88]: is the casting necessary here?

**Review by nastra — APPROVED** [review]

File: core/src/test/java/org/apache/iceberg/TestManifestWriter.java (hunk 1 above)
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/TestManifestWriter.java:88]: nvm, it's necessary

**Comment by tomtongue** [comment]
@nastra Thanks for the quick review!

**Pull request merged**